search-commons/src/main/java/no/unit/nva/search/ticket/TicketAccessFilter.java

```diff
@@ -0,0 +1,276 @@
+        var excludedSet =
+                nonNull(excludeTicketTypes)
+                        ? excludeTicketTypes
+                        : Arrays.stream(TicketType.values())
+                                .filter(f -> !allowedTicketTypes.contains(f))
+                                .collect(
+                                        Collectors.toCollection(
+                                                () -> EnumSet.noneOf(TicketType.class)));
+        return boolQuery()
+                .mustNot(termsQuery(TYPE_KEYWORD, excludedSet))
+                .queryName(NOT_ANY_OF_TICKET_TYPE);
+    }
+
+    private TicketAccessFilter includedTypes(Set<TicketType> ticketTypes) {
+        this.allowedTicketTypes = ticketTypes;
+        return this;
+    }
+
+    private TicketAccessFilter excludeTypes(Set<TicketType> ticketTypes) {
+        this.excludeTicketTypes = ticketTypes;
+        return this;
+    }
+
+    /**
+     * Apply business rules to determine which ticket types are allowed.
+     *
+     * <ul>
+     *   <lh>Business Rules.</lh>
+     *   <li>manage_doi -> DOI_REQUEST
+     *   <li>support -> GENERAL_SUPPORT_CASE
+     *   <li>manage_publishing_requests -> PUBLISHING_REQUEST
+     *   <li>manage_customers -> DOI_REQUEST, GENERAL_SUPPORT_CASE, PUBLISHING_REQUEST
+     *   <li>is_owner -> DOI_REQUEST, GENERAL_SUPPORT_CASE, PUBLISHING_REQUEST
+     * </ul>
+     */
+    private Set<TicketType> applyBusinessRules(List<AccessRight> accessRights)
+            throws UnauthorizedException {
+
+        var isOwner = searchAsTicketOwner() && validateOwner(currentUser);
+        var isCustomManager = searchAsCustomManager() && validateCustomManager(accessRights);
+
+        var allowed = EnumSet.noneOf(TicketType.class);
+        if (accessRights.contains(MANAGE_DOI) || isOwner || isCustomManager) {
+            allowed.add(TicketType.DOI_REQUEST);
+        }
+        if (accessRights.contains(AccessRight.SUPPORT) || isOwner || isCustomManager) {
+            allowed.add(TicketType.GENERAL_SUPPORT_CASE);
+        }
+        if (accessRights.contains(MANAGE_PUBLISHING_REQUESTS) || isOwner || isCustomManager) {
+            allowed.add(TicketType.PUBLISHING_REQUEST);
+        }
+        if (allowed.isEmpty()) {
+            throw new UnauthorizedException(USER_HAS_NO_ACCESS_TO_SEARCH_FOR_ANY_TICKET_TYPES);
+        }
+        return allowed;
+    }
+
+    private Set<TicketType> listToEnumSet(TicketType... ticketTypes) {
+        return Arrays.stream(ticketTypes)
+                .collect(Collectors.toCollection(() -> EnumSet.noneOf(TicketType.class)));
+    }
+
+    private boolean validateCustomManager(List<AccessRight> accessRights)
+            throws UnauthorizedException {
+        if (!accessRights.contains(MANAGE_CUSTOMERS)) {
+            throw new UnauthorizedException(
+                    USER_IS_NOT_ALLOWED_TO_SEARCH_FOR_TICKETS_NOT_OWNED_BY_THEMSELVES);
+        }
+        return true;
+    }
```

Comment:
Rename to hasAccessRight(MANAGE_CUSTOMER) or something similar?

Reply:
`var isOwner = searchAsTicketOwner() && validateOwner(currentUser);`

vs

`
        var isOwner = false; 
        if (searchAsTicketOwner()){
            validateOwner(currentUser);
            isOwner = true;
        }
`

Reply:
When appling accessrights, and you ask for permissions you do not have, we do not continue execution, but throws `UnauthorizedException`

Reply:
Again, I feel like I am missing something

Reply:
It is the naming that is the issue here:

`private boolean validateCustomManager(List<AccessRight> accessRights)`

`customerManager` does not make sense here. `hasAccessToManageCustomers` is something that actually makes sense, and to me it triggers another question. Why would access to manage customers give users access to see other users tickets in the first place? I question why we do this check at all.